Plugins/LicenseCheckerPlugin/main.swift

```diff
@@ -0,0 +1,73 @@
+import Foundation
+import PackagePlugin
+
+@main
+struct LicenseCheckerPlugin: BuildToolPlugin {
+    struct SourcePackagesNotFoundError: Error & CustomStringConvertible {
+        let description: String = "SourcePackages not found"
+    }
+
+    func sourcePackages(_ pluginWorkDirectory: Path) throws -> Path {
+        var tmpPath = pluginWorkDirectory
+        guard pluginWorkDirectory.string.contains("SourcePackages") else {
+            throw SourcePackagesNotFoundError()
+        }
+        while tmpPath.lastComponent != "SourcePackages" {
+            tmpPath = tmpPath.removingLastComponent()
+        }
+        return tmpPath
+    }
+
+    func createBuildCommands(context: PluginContext, target: Target) async throws -> [Command] {
+        let executablePath = try context.tool(named: "LicenseChecker").path
+        let sourcePackagesPath = try sourcePackages(context.pluginWorkDirectory)
+        let whiteListPath = context.package.directory.appending(subpath: "white-list.json")
+
+        return [
+            .buildCommand(
+                displayName: "Check License",
+                executable: executablePath,
+                arguments: [
+                    "--source-packages-path",
+                    sourcePackagesPath.string,
+                    "--white-list-path",
+                    whiteListPath.string
+                ],
+                outputFiles: [
+                    context.pluginWorkDirectory
+                ]
+            )
```

Comment:
Is it not possible to use this as a `prebuildCommand` here? If there are no constraints, we believe it would be more efficient to execute it before the build.
```suggestion
            .prebuildCommand(
                displayName: "Check License",
                executable: executablePath,
                arguments: [
                    "--source-packages-path",
                    sourcePackagesPath.string,
                    "--white-list-path",
                    whiteListPath.string
                ],
                outputFilesDirectory: context.pluginWorkDirectory
            )
```

Reply:
When the `prebuildCommand` is executed, the `LicenseChecker` that BuildToolPlugin depends on has not been built, so it cannot be executed. If you use the `prebuildCommand`, you need to use an artifact bundle that is already built. Using artifact bundles makes version control difficult because the package depends on its own artifacts.

Reply:
Thank you for your explanation. Understood.